src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3626,16 +3631,256 @@ bool isInstanceConstructor(out MethodSymbol constructorSymbol)
             }
         }
 
-        internal virtual BoundExpressionStatement BindConstructorInitializer(ConstructorInitializerSyntax initializer, BindingDiagnosticBag diagnostics)
+#nullable enable
+        [return: NotNullIfNotNull("initializer")]
+        internal virtual BoundExpressionStatement? BindConstructorInitializer(ConstructorInitializerSyntax? initializer, SyntaxNode? ctorSyntax, BindingDiagnosticBag diagnostics)
         {
-            BoundExpression initializerInvocation = GetBinder(initializer).BindConstructorInitializer(initializer.ArgumentList, (MethodSymbol)this.ContainingMember(), diagnostics);
+            Debug.Assert(initializer != null || ctorSyntax != null);
+            BoundExpression? initializerInvocation = initializer is null
+                ? BindImplicitConstructorInitializer((MethodSymbol)this.ContainingMember(), diagnostics, Compilation)
+                : GetRequiredBinder(initializer).BindConstructorInitializer(initializer.ArgumentList, (MethodSymbol)this.ContainingMember(), diagnostics);
+            Debug.Assert(initializerInvocation is not null || initializer is null);
+
+            if (initializerInvocation is null)
+            {
+                return null;
+            }
+
             //  Base WasCompilerGenerated state off of whether constructor is implicitly declared, this will ensure proper instrumentation.
             Debug.Assert(!this.ContainingMember().IsImplicitlyDeclared);
-            var constructorInitializer = new BoundExpressionStatement(initializer, initializerInvocation);
+            var constructorInitializer = new BoundExpressionStatement(initializer ?? ctorSyntax!, initializerInvocation) { WasCompilerGenerated = ((MethodSymbol)ContainingMember()).IsImplicitlyDeclared };
             Debug.Assert(initializerInvocation.HasAnyErrors || constructorInitializer.IsConstructorInitializer(), "Please keep this bound node in sync with BoundNodeExtensions.IsConstructorInitializer.");
             return constructorInitializer;
         }
 
+        /// <summary>
```

Comment:
From here down in this file was a direct copy/paste from `MethodCompiler.cs`, with only minimal changes made to enable nullability.